### PR TITLE
Add Bonsai firmware and via

### PR DESCRIPTION
**Pull request by Retne01**
I've added support for the Bonsai ATmega32A based keyboard.  I've also added VIA support for the Bonsai keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/retne/bonsai/rules.mk [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+F_CPU = 16000000
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/retne/bonsai/rules.mk:14]:
```suggestion
```

File: keyboards/retne/bonsai/rules.mk [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (drashna) [inline keyboards/retne/bonsai/rules.mk:35]:
```suggestion
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
UNICODE_ENABLE = no         # Unicode
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output 
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/retne/bonsai/config.h [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+#pragma once
+
+#include "config_common.h"
+
+#define VENDOR_ID       0x7265  // RE short for Retne
+#define PRODUCT_ID      0x626f  // BO short for Bonsai
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Retne
+#define PRODUCT         Bonsai
+#define DESCRIPTION     Alice style 75% keyboard that can be assembled with only through hole components
```
Comment (drashna) [inline keyboards/retne/bonsai/config.h:26]:
```suggestion
```

File: keyboards/retne/bonsai/readme.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+![Bonsai](https://i.imgur.com/XypHguN.jpg)
+
+A 75% take on the Alice that can be assembled with only through hole components, including usb type-c
+
+* Keyboard Maintainer: [Retne](https://github.com/Retne01)
+* Hardware Supported: Bonsai, atmega32a
+* Hardware Availability: [GitHub](https://github.com/Retne01/Bonsai)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/retne/bonsai/readme.md:11]: Could you add information how to jump to the bootloader?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/retne/bonsai/config.h [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+//#define MAGIC_KEY_LAYER8         8
+//#define MAGIC_KEY_LAYER9         9
+//#define MAGIC_KEY_BOOTLOADER     B
+//#define MAGIC_KEY_BOOTLOADER_ALT ESC
+//#define MAGIC_KEY_LOCK           CAPS
+//#define MAGIC_KEY_EEPROM         E
+//#define MAGIC_KEY_EEPROM_CLEAR   BSPACE
+//#define MAGIC_KEY_NKRO           N
+//#define MAGIC_KEY_SLEEP_LED      Z
+
```
Comment (fauxpark) [inline keyboards/retne/bonsai/config.h:172]:
```suggestion
```

File: keyboards/retne/bonsai/readme.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# Bonsai
+
+![Bonsai](https://i.imgur.com/XypHguN.jpg)
```
Comment (fauxpark) [inline keyboards/retne/bonsai/readme.md:3]:
```suggestion
![Bonsai](https://i.imgur.com/XypHguNh.jpg)
```

File: keyboards/retne/bonsai/rules.mk [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/retne/bonsai/rules.mk:27]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/retne/bonsai/readme.md (hunk 6 above)
Comment (Retne01) [reply keyboards/retne/bonsai/readme.md:3]: Was this change an intended suggestion?  The new suggested url does not point to an image of the keyboard.

Comment (drashna) [reply keyboards/retne/bonsai/readme.md:3]: It does, for me.  The "h" at the end changes which quality that imgur uses for the image.

File: keyboards/retne/bonsai/readme.md (hunk 4 above)
Comment (Retne01) [reply keyboards/retne/bonsai/readme.md:11]: I've got a guide put together on the github for this board.  It covers how to acquire the parts, solder the board together, how to flash the bootloader to the MCU, and finally how to flash QMK to the MCU.  Would it be worth linking to it here?  If not should I include how to do it after QMK has been loaded onto the keyboard or before?  I included the reset command on Fn+R but the board can also be reset by holding down the boot button and tapping the reset button.

File: keyboards/retne/bonsai/readme.md (hunk 6 above)
Comment (Retne01) [reply keyboards/retne/bonsai/readme.md:3]: Ah interesting.  I looked at this on my phone before I sat down at the PC and the url didn't go anywhere on my phone.  I'll commit it.

File: keyboards/retne/bonsai/readme.md (hunk 4 above)
Comment (drashna) [reply keyboards/retne/bonsai/readme.md:11]: Ideally, a short bit about how to jump to bootloader should be included in the readme here.

Or at least, a link to the repo with the docs.

File: keyboards/retne/bonsai/config.h [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 1
+
```
Comment (fauxpark) [inline keyboards/retne/bonsai/config.h:159]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/retne/bonsai/readme.md (hunk 4 above)
Comment (drashna) [reply keyboards/retne/bonsai/readme.md:11]: Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Comment by peepeetee** [comment]
Hi, any update to this?

File: keyboards/retne/bonsai/rules.mk [hunk 9]
```diff
@@ -0,0 +1,25 @@
+# MCU name
+MCU = atmega32a
+
+# Processor frequency
+F_CPU = 16000000
+
+# Bootloader selection
+BOOTLOADER = USBasp
```
Comment (fauxpark) [inline keyboards/retne/bonsai/rules.mk:8]:
```suggestion
BOOTLOADER = usbasploader
```

File: keyboards/retne/bonsai/rules.mk [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/retne/bonsai/rules.mk:13]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/retne/bonsai/rules.mk [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/retne/bonsai/rules.mk:21]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/retne/bonsai/rules.mk [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/retne/bonsai/rules.mk:24]:
```suggestion
```

File: keyboards/retne/bonsai/info.json [hunk 13]
```diff
@@ -0,0 +1,15 @@
+{
+    "keyboard_name": "Bonsai",
+    "url": "https://github.com/Retne01/Bonsai",
+    "maintainer": "Retne",
+    "width": 18.25,
+    "height": 10.55,
```
Comment (fauxpark) [inline keyboards/retne/bonsai/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**